### ISSUE TITLE
Tests don't pass for some solutions

### ISSUE THREAD

**Issue by sheriffderek** [FrontendMasters/api-design-node-v3#12]
When following along, my tests didn't pass for lesson 2. It was a small exercise - so, I checked the solution and everything was correct. I made sure I pulled dependencies sense there are a few things added in-between each step. Then I switched to the solution branch pulled dependencies / and ran the tests - but they didn't pass there either.

I'm not sure why it would be interacting with a database yet (at exercise 2) (so that shouldn't matter) - but I got some errors related to Mongo and wondering how that might fit in.

I tried both - `yarn test-routes` and `npm run test-routes` commands in case there was a discrepancy.

I jumped between a few lessons and they didn't pass either.

Am I doing something wrong? Or has something changed - that I don't know about? 

RECREATE: pull `lesson-2-solution`: update dependencies: run tests

`.get(controllers.getOne)`  >   `getOne` type functions aren't written in the controller - so, that fails for each verb. Adding them / or just tossing in a callback solves for those - 

but then:

(node:9940) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'dropDatabase' of undefined
(node:9940) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:9940) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
 FAIL  src/resources/item/__tests__/item.router.spec.js
  item router
    ✕ has crud routes (80ms)

  ● item router › has crud routes

    MongoNetworkError: failed to connect to server [localhost:27017] on first connect [MongoNetworkError: getaddrinfo ENOTFOUND localhost localhost:27017]

      at Pool.<anonymous> (node_modules/mongodb-core/lib/topologies/server.js:564:11)
      at Connection.<anonymous> (node_modules/mongodb-core/lib/connection/pool.js:317:12)
      at Socket.<anonymous> (node_modules/mongodb-core/lib/connection/connection.js:246:50)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 1 total

**Comment by aamnah** [comment]
Same for me, can't run either `npm run dev`  or `npm run test-routes`. 

`npm run dev` gives a `MongoNetworkError`

```
npm run dev

> api-design-v3@1.0.0 dev /mnt/r/tmp/api-design-node-v3
> nodemon --exec npm run restart

[nodemon] 1.18.9
[nodemon] reading config ./nodemon.json
[nodemon] to restart at any time, enter `rs`
[nodemon] or send SIGHUP to 8469 to restart
[nodemon] ignoring: ./.git/**/* node_modules/**/node_modules ./dist/**/*
[nodemon] watching: src/**/*.js src/**/*.graphql src/**/*.gql
[nodemon] watching extensions: js,json,graphql
[nodemon] starting `npm run restart`
[nodemon] spawning
[nodemon] child pid: 8482
[nodemon] watching 24 files

> api-design-v3@1.0.0 restart /mnt/r/tmp/api-design-node-v3
> rimraf dist && npm run build && npm run start


> api-design-v3@1.0.0 build /mnt/r/tmp/api-design-node-v3
> babel src --out-dir dist

Successfully compiled 24 files with Babel.

> api-design-v3@1.0.0 start /mnt/r/tmp/api-design-node-v3
> node dist/index.js

{ MongoNetworkError: failed to connect to server [localhost:27017] on first connect [MongoNetworkError: connect ECONNREFUSED 127.0.0.1:27017]
    at Pool.<anonymous> (/mnt/r/tmp/api-design-node-v3/node_modules/mongodb-core/lib/topologies/server.js:564:11)
    at Pool.emit (events.js:189:13)
    at Connection.<anonymous> (/mnt/r/tmp/api-design-node-v3/node_modules/mongodb-core/lib/connection/pool.js:317:12)
    at Object.onceWrapper (events.js:277:13)
    at Connection.emit (events.js:189:13)
    at Socket.<anonymous> (/mnt/r/tmp/api-design-node-v3/node_modules/mongodb-core/lib/connection/connection.js:246:50)
    at Object.onceWrapper (events.js:277:13)
    at Socket.emit (events.js:189:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  name: 'MongoNetworkError',
  errorLabels: [ 'TransientTransactionError' ],
  [Symbol(mongoErrorContextSymbol)]: {} }
[nodemon] clean exit - waiting for changes before restart
```

and `npm run test-routes` gives me a `bcrypt` error, even though i have run `npm install` in this branch

```
npm run dev

> api-design-v3@1.0.0 dev /mnt/r/tmp/api-design-node-v3
> nodemon --exec npm run restart

[nodemon] 1.18.9
[nodemon] reading config ./nodemon.json
[nodemon] to restart at any time, enter `rs`
[nodemon] or send SIGHUP to 8469 to restart
[nodemon] ignoring: ./.git/**/* node_modules/**/node_modules ./dist/**/*
[nodemon] watching: src/**/*.js src/**/*.graphql src/**/*.gql
[nodemon] watching extensions: js,json,graphql
[nodemon] starting `npm run restart`
[nodemon] spawning
[nodemon] child pid: 8482
[nodemon] watching 24 files

> api-design-v3@1.0.0 restart /mnt/r/tmp/api-design-node-v3
> rimraf dist && npm run build && npm run start


> api-design-v3@1.0.0 build /mnt/r/tmp/api-design-node-v3
> babel src --out-dir dist

Successfully compiled 24 files with Babel.

> api-design-v3@1.0.0 start /mnt/r/tmp/api-design-node-v3
> node dist/index.js

{ MongoNetworkError: failed to connect to server [localhost:27017] on first connect [MongoNetworkError: connect ECONNREFUSED 127.0.0.1:27017]
    at Pool.<anonymous> (/mnt/r/tmp/api-design-node-v3/node_modules/mongodb-core/lib/topologies/server.js:564:11)
    at Pool.emit (events.js:189:13)
    at Connection.<anonymous> (/mnt/r/tmp/api-design-node-v3/node_modules/mongodb-core/lib/connection/pool.js:317:12)
    at Object.onceWrapper (events.js:277:13)
    at Connection.emit (events.js:189:13)
    at Socket.<anonymous> (/mnt/r/tmp/api-design-node-v3/node_modules/mongodb-core/lib/connection/connection.js:246:50)
    at Object.onceWrapper (events.js:277:13)
    at Socket.emit (events.js:189:13)
    at emitErrorNT (internal/streams/destroy.js:82:8)
    at emitErrorAndCloseNT (internal/streams/destroy.js:50:3)
    at process._tickCallback (internal/process/next_tick.js:63:19)
  name: 'MongoNetworkError',
  errorLabels: [ 'TransientTransactionError' ],
  [Symbol(mongoErrorContextSymbol)]: {} }
[nodemon] clean exit - waiting for changes before restart
```

**Comment by laniehei** [comment]
I was able to recreate the error. Please make sure you the mongo database running by [installing it](https://docs.mongodb.com/manual/administration/install-community/), running the command `mongod` in another terminal window, then make sure the following code is [in your repository](https://github.com/FrontendMasters/api-design-node-v3/blob/master/src/resources/item/item.controllers.js)

**Comment by Kolombiken** [comment]
@laniehei Wouldn't it be better if [the things for Mongo in before each test and after each test](https://github.com/FrontendMasters/api-design-node-v3/blob/lesson-2/test-db-setup.js) were removed in the branch for Lesson 2? 
I've followed the videos closely and as far as I can tell there is nothing about setting up MongoDB before running the test at this moment in the course. 

I'm aware that I'm commenting on a closed issue but I thought it was better than opening a new one.
